### PR TITLE
Feature/us159246

### PR DESCRIPTION
**Pull request by debrin182**
Reimpement show relations mashup
1) Add support for Timeline
2) Drag-n-drop is supported with shown relations
3) Legend is shown in Actions menu
4) Reduce coupling with TP

### CODE REVIEW

File: src/utils/relation.lines.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    const cos = Math.cos(rad);
+    const sin = Math.sin(rad);
+
+    const centerOnLine = {
+        x: (start.x + end.x) / 2,
+        y: (start.y + end.y) / 2
+    };
+
+    const centerRot = {
+        x: (centerOnLine.x - start.x) * cos - (centerOnLine.y - start.y) * sin + start.x,
```
Comment (dmitryTsarevich) [inline src/utils/relation.lines.js]: Can be simplified:
```
x: (centerOnLine.x - start.x) * cos - (centerOnLine.y - start.y) * sin + start.x
=>
x: ((start.x + end.x) / 2 - start.x) * cos - ((start.y + end.y) / 2 - start.y) * sin + start.x  // substitute centerOnLine
=>
x: ((end.x - start.x) / 2) * cos - ((end.y - start.y) / 2) * sin + start.x  // open arrows, compute, close
```

Do the same for `y` and remove `centerOnLine` entirely.

File: src/utils/relation.lines.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+export const getInboundRelationTypeMarkerEndId = ({name}) => `${name}_inbound_end`;
+export const getOutboundRelationTypeMarkerEndId = ({name}) => `${name}_outbound_end`;
+export const getRelationTypeColor = ({style}) => style;
+export const getRelationColor = (relation) =>
+    getRelationTypeColor(relationTypes.filter(({name}) => name === relation)[0]);
+export const generateBezierCoords = (start, end, down = false) => {
+
+    let points = [`M${start.x},${start.y}`];
+
+    const rad = Math.PI / 48 * (down ? -1 : 1);
```
Comment (dmitryTsarevich) [inline src/utils/relation.lines.js]: Precompute this like:
```
const rad = Math.PI / 48;
const cos = Math.cos(rad);   // cos(Math.PI / 48) === cos(Math.PI / -48)
const positiveSin = Math.sin(rad);     
const sines = [positiveSin, -positiveSin];   // -sin(Math.PI / 48) === sin(Math.PI / -48)
....
export const generateBezierCoords = (...) {
     use precomputed cos for cos.
     const sin = sines[down];   // select correct sin value.

File: src/utils/intersection.js [hunk 3]
```diff
@@ -40,13 +40,15 @@ const getSlicesByRect = (rect) => {
...
 
-    const isY1 = point.y >= Math.min(s1.y1, s1.y2) && point.y <= Math.max(s1.y1, s1.y2);
-    const isY2 = point.y >= Math.min(s2.y1, s2.y2) && point.y <= Math.max(s2.y1, s2.y2);
+const checkInSlices = (point, s1, s2) => {
+    const isX1 = isBetween(point.x, Math.min(s1.x1, s1.x2), Math.max(s1.x1, s1.x2));
+    const isX2 = isBetween(point.x, Math.min(s2.x1, s2.x2), Math.max(s2.x1, s2.x2));
+    const isY1 = isBetween(point.y, Math.min(s1.y1, s1.y2), Math.max(s1.y1, s1.y2));
+    const isY2 = isBetween(point.y, Math.min(s2.y1, s2.y2), Math.max(s2.y1, s2.y2));
 
     return isX1 && isX2 && isY1 && isY2;
```
Comment (dmitryTsarevich) [inline src/utils/intersection.js]: Lazy compute isX2 and below:
```
return isBetween(point.x, Math.min(s1.x1, s1.x2), Math.max(s1.x1, s1.x2)) &&  // x1
   isBetween(point.x, Math.min(s2.x1, s2.x2), Math.max(s2.x1, s2.x2)) && // x2
   ....
```

File: src/const.view.modes.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Enum for view modes values.
+ * @readonly
+ * @enum {string}
+ */
+const viewModes = {
+    LIST: 'newlist',
```
[the remark this answers is not in the document]
Comment (debrin182) [reply src/const.view.modes.js]: Good point, but even inside TP this view modes used with a string literals across multiple files. It requires to review the whole codebase :(

File: src/data.js [hunk 5]
```diff
@@ -18,58 +19,114 @@ const loadPages = (url, params) =>
...
+                    this.subscribeForRelationsChanged();
+                }
+            }
+        });
+    }
+
+    load(entityIds) {
+        if (_.intersection(entityIds, this.entityIds).length === entityIds.length) {
+            return Promise.resolve(this.getRelationsFiltered());
+        } else {
```
[the remark this answers is not in the document]
Comment (debrin182) [reply src/data.js]: It's a point to discuss. When branching is explicit, i'd rather use else statement. In this case only one of two options is possible, so else statement underlines this explicitly.

**Pull request merged**